### DISCUSSION TITLE
How to fetch data within a radius of xx KM (or MI) from a location ?

### DISCUSSION THREAD

**Discussion by SanNic20** [sequelize/sequelize#14717] [Help & Questions] [answered]
Hello everyone and thank you in advance for your precious help.

I'm looking for a way to fetch data within a radius of xx KM (or MI) from a location ? But I don't see how to do it with Sequelize and PostgreSQL :(

I specify that I "simply" want to find the objects that are within a radius, no need to sort on the nearest object.

Currently the model of my objects is as below :

```
const item = (sequelize, DataTypes) => {
  const Item = sequelize.define('item', {
    title: {
      type: DataTypes.STRING,
      unique: false,
      allowNull: false,
      validate: {
        notEmpty: true,
        len: [1, 50],
      },
    },
    description: {
      type: DataTypes.STRING,
      unique: false,
      allowNull: false,
      validate: {
        notEmpty: true,
        len: [1, 200],
      },
    },
    latitude: {
      type: DataTypes.FLOAT,
      unique: false,
      allowNull: true,
      validate: {
        notEmpty: false,
      },
    },
    longitude: {
      type: DataTypes.FLOAT,
      unique: false,
      allowNull: true,
      validate: {
        notEmpty: false,
      },
    },
  },
  );
```
First, can I do this with my data model ? Or do I have to put the location (latitude and longitude) in a "DataTypes.GEOMETRY('POINT')" ? 

And lastly, how to do the query condition to achieve that ? I know that with mongoose, there are easy ways to do it with "$geoWithin" and "$centerSphere", but with Sequelize I don't have too much knowledge to get this easily.

Any help will be much appreciated, thanks again in advance for your help.

**Comment by doneumark** [comment, marked as answer]
Hey,
This is the code I use:
```
where: sequelize.where(
	sequelize.fn(
		'ST_DistanceSphere',
		sequelize.col('coordinates'),
		sequelize.literal(`ST_GeomFromText('POINT(${someLongitude} ${someLatitude})')`),
	),
	{ [Op.lt]: maxRadius },
)
```

Where the 'coordinates' column is a `DataTypes.GEOMETRY('POINT')` type, `someLongitude` and `someLatitude` holds the coordinates of the desired location. And `maxRadius` is the allowed distance from the location.

**Reply by SanNic20** [reply]
@doneumark : Great, thank you very much for your answer and sharing your code. Everything is much simpler now ;) Wish you a nice summer.

**Reply by ephys** [reply]
Concatenating values in a literal like this can lead to SQL injections, use a replacement for `` `POINT(${someLongitude} ${someLatitude})` `` instead:

```js
MyModel.findAll({
  where: where(
    fn(
      'ST_DistanceSphere',
      col('coordinates'),
      literal(`ST_GeomFromText(:coordinates)`),
    ),
    { [Op.lt]: maxRadius },
  ),
  replacements: {
    coordinates: `POINT(${someLongitude} ${someLatitude})`,
  },
})

```

**Reply by SanNic20** [reply]
@ephys : Thank you for this valuable information and for adapting the source code
